src/main/java/edu/ucsb/cs156/happiercows/controllers/StudentsController.java

```diff
@@ -0,0 +1,121 @@
+@Slf4j
+public class StudentsController extends ApiController {
+
+    @Autowired
+    StudentsRepository StudentsRepository;
+
+    /**
+     * List all students
+     * 
+     * @return an iterable of Students
+     */
+    @Operation(summary= "List all students")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("/all")
+    public Iterable<Students> allStudents() {
+        Iterable<Students> students = StudentsRepository.findAll();
+        return students;
+    }
+
+    /**
+     * Create new Student
+     * 
+     * @param lName        last name
+     * @param fmName       first and middle name
+     * @param email           email
+     * @param perm            Perm number
+     * @param courseId        course ID
+     * @return saved student
+     */
+    @Operation(summary= "Create a Student")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping("/post")
+    public Students postStudents(
+            @Parameter(name = "lName") @RequestParam String lName,
+            @Parameter(name = "fmName") @RequestParam String fmName,
+            @Parameter(name = "email") @RequestParam String email,
+            @Parameter(name = "perm") @RequestParam String perm,
+            @Parameter(name="courseId") @RequestParam Long courseId)
+            throws JsonProcessingException {
+        Students student = Students.builder()
+                .lName(lName)
+                .fmName(fmName)
+                .email(email)
+                .perm(perm)
+                .courseId(courseId)
+                .build();
+        Students savedStudents = StudentsRepository.save(student);
+        return savedStudents;
+    }
+
+    /**
+     * Get a student by id
+     * 
+     * @param id the id of the student
+     * @return a Student
+     */
+    @Operation(summary = "Get a single Student by id")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("")
+    public Students getById(
+            @Parameter(name = "id") @RequestParam Long id) {
+        Students students = StudentsRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(Students.class, id));
+
+        return students;
+    }
+
+        /**
+     * Update a single Student
+     * 
+     * @param id       id of the Student to update
+     * @param incoming the new Student
+     * @return the updated Students object
+     */
+    @Operation(summary= "Update a single Students")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PutMapping("")
+    public Students updateStudents(
+            @Parameter(name="id") @RequestParam Long id,
+            @RequestBody @Valid Students incoming) {
+
+        Students student = StudentsRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(Students.class, id));
+
+        student.setLName(incoming.getLName());
+        student.setFmName(incoming.getFmName());
+        student.setEmail(incoming.getEmail());
+        student.setCourseId(incoming.getCourseId());
```

Comment:
We shouldn't mess with the course id.  Once a student is added to a course, we should not change the course id.  If they were added to the wrong course they should be deleted from the one course and added to another one.

The reasons for that are that later functionality may depend on a student object being "immutable" from the standpoint of which course they "belong to" (including, for example, adding an "Instructor" role in between ADMIN and USER that can only work with the Courses and Commons that they themselves created.